packages/react-native-sdk/src/providers/StreamCall/RejectCallWhenBusy.tsx

```diff
@@ -0,0 +1,73 @@
+import { useEffect } from 'react';
+import {
+  useCalls,
+  useStreamVideoClient,
+} from '@stream-io/video-react-bindings';
+import { CallingState, getLogger } from '@stream-io/video-client';
+import InCallManager from 'react-native-incall-manager';
+import { StreamVideoRN } from '../../utils';
+import { Platform } from 'react-native';
+
+/**
+ * This is a renderless component to reject calls when the user is busy.
+ */
+export const RejectCallWhenBusy = () => {
+  const client = useStreamVideoClient();
+  const calls = useCalls();
+
+  useEffect(() => {
+    if (!client) return;
+    return client?.on('call.rejected', async (event) => {
+      // Workaround needed for the busy tone:
+      // This is because the call was rejected without even starting,
+      // before calling the stop method with busy tone we need to start the call first.
+      InCallManager.start({ media: 'audio' });
+
+      const callCid = event.call_cid;
+      const callId = callCid.split(':')[1];
+      if (!callId) return;
+
+      const rejectedCall = client?.call(event.call.type, callId);
+      await rejectedCall?.getOrCreate();
```

Comment:
The `event` already contains the `call` data. I think we can get rid of this `getOrCreate()`

```ts
event.call.created_by.id === client.state.connectedUser
```